File: golang/09-stringly-typed/issue_updater_test.go
package kata

import (
	"strings"
	"testing"
)

func TestUpdateIssueClosesIssue(t *testing.T) {
	iu := NewIssueUpdater()
	result := iu.update_issue(42, "close")
	if !strings.Contains(result, "status=closed") {
		t.Errorf("expected result to contain 'status=closed', got %q", result)
	}
}

func TestUpdateIssueOpensIssue(t *testing.T) {
	iu := NewIssueUpdater()
	result := iu.update_issue(42, "open")
	if !strings.Contains(result, "status=open") {
		t.Errorf("expected result to contain 'status=open', got %q", result)
	}
}

func TestUpdateIssueSetsInProgress(t *testing.T) {
	iu := NewIssueUpdater()
	result := iu.update_issue(42, "progress")
	if !strings.Contains(result, "status=in_progress") {
		t.Errorf("expected result to contain 'status=in_progress', got %q", result)
	}
}

func TestUpdateIssueSetsPriority1(t *testing.T) {
	iu := NewIssueUpdater()
	result := iu.update_issue(42, "close:1")
	if !strings.Contains(result, "priority=1") {
		t.Errorf("expected result to contain 'priority=1', got %q", result)
	}
}

func TestUpdateIssueSetsPriority2(t *testing.T) {
	iu := NewIssueUpdater()
	result := iu.update_issue(42, "open:2")
	if !strings.Contains(result, "priority=2") {
		t.Errorf("expected result to contain 'priority=2', got %q", result)
	}
}

func TestUpdateIssueDefaultsToPriority3(t *testing.T) {
	iu := NewIssueUpdater()
	result := iu.update_issue(42, "progress")
	if !strings.Contains(result, "priority=3") {
		t.Errorf("expected result to contain 'priority=3', got %q", result)
	}
}

func TestUpdateIssueIncludesIssueId(t *testing.T) {
	iu := NewIssueUpdater()
	result := iu.update_issue(99, "close")
	if !strings.Contains(result, "Issue 99") {
		t.Errorf("expected result to contain 'Issue 99', got %q", result)
	}
}

func TestUpdateIssueIgnoresInvalidPriority(t *testing.T) {
	iu := NewIssueUpdater()
	result := iu.update_issue(42, "close:99")
	if !strings.Contains(result, "priority=3") {
		t.Errorf("expected result to contain 'priority=3', got %q", result)
	}
}

File: csharp/09-stringly-typed/test/issue_updater_test.cs
using Xunit;

public class IssueUpdaterTest
{
    [Fact]
    public void ClosesIssue()
    {
        var updater = new IssueUpdater();
        var result = updater.update_issue(42, "close");
        Assert.Contains("status=closed", result);
    }

    [Fact]
    public void OpensIssue()
    {
        var updater = new IssueUpdater();
        var result = updater.update_issue(42, "open");
        Assert.Contains("status=open", result);
    }

    [Fact]
    public void SetsInProgress()
    {
        var updater = new IssueUpdater();
        var result = updater.update_issue(42, "progress");
        Assert.Contains("status=in_progress", result);
    }

    [Fact]
    public void SetsPriority1()
    {
        var updater = new IssueUpdater();
        var result = updater.update_issue(42, "close:1");
        Assert.Contains("priority=1", result);
    }

    [Fact]
    public void SetsPriority2()
    {
        var updater = new IssueUpdater();
        var result = updater.update_issue(42, "open:2");
        Assert.Contains("priority=2", result);
    }

    [Fact]
    public void DefaultsToPriority3()
    {
        var updater = new IssueUpdater();
        var result = updater.update_issue(42, "progress");
        Assert.Contains("priority=3", result);
    }

    [Fact]
    public void IncludesIssueId()
    {
        var updater = new IssueUpdater();
        var result = updater.update_issue(99, "close");
        Assert.Contains("Issue 99", result);
    }

    [Fact]
    public void IgnoresInvalidPriority()
    {
        var updater = new IssueUpdater();
        var result = updater.update_issue(42, "close:99");
        Assert.Contains("priority=3", result);
    }
}
